### PR TITLE
Don't pass reference to mutable slice

### PR DESCRIPTION
**Pull request by cpuguy83**

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]
LGTM

It is sad that go lacks immutable list 😭

File: pkg/registrar/registrar.go [hunk 1]
```diff
@@ -98,7 +98,12 @@ func (r *Registrar) GetNames(key string) ([]string, error) {
 	if !exists {
 		return nil, ErrNoSuchKey
 	}
-	return names, nil
+
+	ls := make([]string, 0, len(names))
+	for _, n := range names {
```
Comment (ripcurld0) [inline pkg/registrar/registrar.go]: LGTM but just curious why not doing so with `ls = append(ls, names...)` or `copy(ls, names)`?

**Review by vdemeester — APPROVED** [review]
LGTM 🦁

File: pkg/registrar/registrar.go (hunk 1 above)
Comment (cpuguy83) [inline pkg/registrar/registrar.go]: No real reason other than I've had the results of `copy` suprise me more than once.

**Pull request merged**